lib/open_project/authentication/strategies/warden/global_basic_auth.rb

```diff
@@ -0,0 +1,82 @@
+require 'warden/basic_auth'
+
+module OpenProject
+  module Authentication
+    module Strategies
+      module Warden
+        ##
+        # Allows authentication via a singular set of basic auth credentials for admin access.
+        #
+        # The credentials must be configured in `config/configuration.yml` like this:
+        #
+        #     production:
+        #       authentication:
+        #         global_basic_auth:
+        #           user: admin
+        #           password: 123456
+        #
+        # The strategy will only be triggered when the configured user name is sent.
+        # Meaning that this strategy is skipped if a basic auth attempt involving any
+        # other user name is made.
+        class GlobalBasicAuth < ::Warden::Strategies::BasicAuth
+          def self.configuration
+            @configuration ||= configure!
+          end
+
+          ##
+          # Updates the configuration for this strategy. It's usually called only once, at startup.
+          #
+          # @param [Hash] config The configuration to be used. Must contain :user and :password.
+          # @raise [ArgumentError] Raises an error if the configured user name collides with the
+          #                        user name used for UserBasicAuth (apikey).
+          def self.configure!(config = openproject_config)
+            if config[:user] == UserBasicAuth.user
+              raise ArgumentError, "global user must not be '#{UserBasicAuth.user}'"
+            end
+
+            if config[:password].blank?
+              raise ArgumentError, "password must not be empty"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.